design/benchmark_tests.md

```diff
@@ -0,0 +1,33 @@
+# Benchmarks Tests
+
+## Requirements
+1. Measure the latency percentiles (50th, 90th, 95th, 99th 99.9) for a batch of repeated basic operations (e.g. 10K put requests),
+   at a predefined parallelism level.
+2. Show trends of benchmarks tests execution times across different versions.
+3. Portability: The tests don't assume anything on the environment where the lakeFS & DB are running.
+   lakeFS address is the only **required** argument for running the tests.
+
+## Non-Requirements
+1. An active gatekeeper - single slow running cycle shouldn't block the build.
+2. Expect 100% reliability of all actions - when performing thousands of operations,
+   some (exact percent TBD) will be allowed to fail. When measuring an external app,
+   it's the app responsibility to retry on transient errors.
+3. Avoid repeating unit/integration/system tests scenarios.
+  
+## Solution
+### Test Execution
+For each parallelism level, the flow will create a repository and will PUT 10k files in it.
+Then, it will try to read those files. Finally, it will delete them.
+
+### CI
+Github workflow is triggered on a merge to master.
+The workflow will deploy a temporary environment that includes the latest lakeFS app, and an RDS instance.
+Workflow collects and structures the tests results from the lakeFS instance, 
+and sends them to a dedicated `prometheus` server.
+The deployed environment is deleted after each collection of the results. 
+
+### Results
+A `Grafana` dashboard will display the benchmark tests results
```

Comment:
If all we care about is success/error rate and latency distribution, we can generate a report and stick it in a PR comment. Or any other form that will allow us to track this over time over different versions without relying on a specific Grafana availability. With the added benenfit of being able to look at a specific merge to master and see its stats without trying to align that with Grafana's time based view of the world.

Reply:
Not a bad idea for catching a single PR that introduce a degradation, but how do I see how it changes over time?